pathways/inspections.py

```diff
@@ -68,65 +67,212 @@ def inspect_first_n(num_boxes, shipment):
     return True, num_boxes
 
 
-def inspect_shipment_percentage(config, shipment):
-    """Inspect shipments based on the percetantage strategy
+def sample_percentage(config, shipment):
+    """Set sample size to sample units from shipment using percentage strategy.
+    Convert to units to boxes if needed.
+    Return number of boxes to inspect.
+
+    :param config: Configuration to be used595
+    :param shipment: Shipment to be inspected
+    """
+    unit = config["inspection"]["unit"]
+    ratio = config["inspection"]["percentage"]["proportion"]
+    num_stems = shipment["num_stems"]
+    num_boxes = shipment["num_boxes"]
+
+    if unit == "stems":
+        n_stems_to_inspect = int(math.ceil(ratio * num_stems))
+        n_boxes_to_inspect = convert_stems_to_boxes(config, shipment, n_stems_to_inspect)
+    elif unit =="boxes":
+        n_boxes_to_inspect = int(math.ceil(ratio * num_boxes))
+        n_boxes_to_inspect = max(min_boxes, n_units_to_inspect)
+        n_boxes_to_inspect = min(num_boxes, n_boxes_to_inspect)
+    else:
+        raise RuntimeError(
+            "Unknown sampling unit: {unit}".format(**locals())
+        )
+    return n_boxes_to_inspect
+
+
+def sample_hypergeometric(config, shipment):
+    """Set sample size to sample units from shipment using hypergeometric/detection
+    level strategy. Convert to units to boxes if needed.
+    Return number of boxes to inspect.
 
     :param config: Configuration to be used
-    :param shipement: Shipment to be inspected
+    :param shipment: Shipment to be inspected
     """
-    ratio = config["proportion"]
-    min_boxes = config.get("min_boxes", 1)
-    # closest higher integer
-    boxes_to_inspect = int(math.ceil(ratio * len(shipment["boxes"])))
-    boxes_to_inspect = max(min_boxes, boxes_to_inspect)
-    boxes_to_inspect = min(len(shipment["boxes"]), boxes_to_inspect)
-    # in any case, first n boxes
-    strategy = config["end_strategy"]
-    if strategy == "to_completion":
-        pest = 0
-        for i in range(boxes_to_inspect):
-            if shipment["boxes"][i]:
-                pest += 1
-        return pest == 0, boxes_to_inspect
-    elif strategy == "to_detection":
-        for i in range(boxes_to_inspect):
-            if shipment["boxes"][i]:
-                return False, i + 1
-        return True, boxes_to_inspect
+    unit = config["inspection"]["unit"]
+    detection_level = config["inspection"]["hypergeometric"]["detection_level"]
+    confidence_level = config["inspection"]["hypergeometric"]["confidence_level"]
+    num_stems = shipment["num_stems"]
+    num_boxes = shipment["num_boxes"]
+
+    if unit =="stems":
+        n_stems_to_inspect = math.ceil((1-((1-confidence_level)**
+        (1/(detection_level*num_stems))))*(num_stems-(((detection_level*num_stems)-1)/2)))
```

Comment:
This looks like a candidate for a function. It is complicated and it is there twice. If you have a(ny) name for it, just go ahead and make it a function.

Reply:
Added function.